runtime/src/gpu/nvidia/gpu-nvidia.c

```diff
@@ -104,8 +112,10 @@ void chpl_gpu_impl_load_global(const char* global_name, void** ptr,
   CUmodule module;
 
   CUDA_CALL(cuCtxGetDevice(&device));
-  module = chpl_gpu_cuda_modules[(int)device];
-
+  assert((((int) device) >= 0) && (((int) device) < numAllDevices));
+  int index = deviceIDToIndex[(int)device];
+  assert((index >= 0) && (index < numDevices));
+  module = chpl_gpu_cuda_modules[index];
```

Comment:
Consider making this a helper method to avoid the duplication

Reply:
Fixed.